lib/abstract_client.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const Beanstalk = require('fivebeans').client;
+const moment = require('moment');
+const bb = require('bluebird');
+const _ = require('lodash');
+
+/**
+ * Abstract beanstalk client class
+ */
+class AbstractClient {
+	constructor(config) {
```

Comment:
Assume previous implementation in correct, It support multiple job reserve.
In the version will change to only reserve and process 1 job?

Reply:
@eric4545 correct, original implementation supports multiple jobs reserve